### PR TITLE
Add contributor guide

### LINKED ISSUE #458: Add CONTRIBUTING.md
The particular items to include so far:

* License agreement.
* Explanation of labels used on issues and pull requests.
* Explanation regarding the importance of open discussion on code, documentation, issues, features, bugs, and pull requests by *any contributor* at any time.
* Explanation of the use of certain emoji in discussions. In particular:
  * :question: Indicates a question. This emoji comes with "no strings attached", which means it's fine to simply answer the question in another comment without making any changes to code.
  * :exclamation: Indicates a problem with the code which will need to be addressed before changes can be merged into `master`. This *may* be accompanied by a specific suggestion for how to change the code. Any contributor is completely welcome to open a discussion regarding alternative solutions, even if the originally proposed change has already been made.
  * :bulb: Indicates an idea or suggestion regarding a potential change in the code. These items will not necessarily block the inclusion of code into the `master` branch. In certain trivial cases (e.g. a simple formatting change), a core contributor on the project may make the suggested change themselves in the process of merging a pull request. Like the :exclamation: emoji, any suggestion included with the :bulb: is open for discussion by any contributor.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #458

@rackerlabs/docs-devdoc Can you review this contributor guide for tone and clarity? :smiley:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Contributor Guide
+
+## Open Discussion
+
+The most important part of an open source project is *open discussion*. If a developer, user, or other contributor has
+a question, comment, suggestion, or concern about any part of this project, **please "speak up"** (create an issue or
+add a comment to an existing issue or pull request).
+
+This policy especially affects the emoji described below. Any user or contributor is free to use these symbols to
+clarify intent during discussions. Questions and concerns from first-time contributors will be considered just as
```
Comment (catlook) [inline CONTRIBUTING.md]: ... Any user or contributor may freely use these symbols ...
... concerns from first-time contributors are considered ....

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+important as questions and concerns from the original project developers.
+
+## License
+
+This project is licensed under the MIT License. A complete description of the projects' license, as well as license
+notices for 3rd party code incorporated into this project, is available in the repository:
```
Comment (catlook) [inline CONTRIBUTING.md]: > ... Find a complete description of the project's license, as well as license notices for 3rd party code incorporated into this project, in the repository ...

Comment (sharwell) [inline CONTRIBUTING.md]: What about this?

> The code in this repository is licensed under the MIT License. A complete description of the license, as well as ...

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+| duplicate | Another issue or pull request contains the original report for this topic | Another pull request was submitted to correct the issue. This is generally only applied to pull requests after another pull request to correct the issue is merged. |
+| wontfix | The issue will not be corrected. The current behavior could be by design, out of scope, or cannot be changed due to the breaking changes policy for the project (see comments for details). | The pull request will not be merged due to a fundamental issue (see description for this label on issues). |
+| v1 | The issue affects the openstack.net V1.x SDK | The pull request affects the openstack.net V1.x SDK |
+| v2 | The issue affects the OpenStack.NET V2.x SDK | The pull request affects the OpenStack.NET V2.x SDK |
+
+## Emoji
+
+GitHub provides several emoji which can be included in many areas of the site, especially including comments on issues
+and pull requests. To reduce confusion during evaluation of a pull request, the following Emoji can be used to convey
```
Comment (sharwell) [inline CONTRIBUTING.md]: Emoji should not be capitalized here

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  `master`. This *may* be accompanied by a specific suggestion for how to change the code. Any contributor is completely
+  welcome to open a discussion regarding alternative solutions, even if the originally proposed change has already been
+  made.
+
+  Every active contributor to a project is likely to use the :exclamation: emoji (or similar) due to a simple
+  misunderstanding or oversight in their evaluation. **This is absolutely fine.** If you notice a case like this, simply
+  add a comment to clarify the intent and the exclamation can be considered resolved.
+
+  In general, pull requests with an unresolved exclamation will not be merged until they are addressed. While it is
+  possible for code with an exclamation to be merged, this will generally only occur in cases where the rationale is
```
Comment (catlook) [inline CONTRIBUTING.md]: I'd bullet point the three cases where code can be merged.

File: README.md [hunk 5]
```diff
@@ -1,16 +1,20 @@
...
-<p>The openstack.net SDK contains:
-	<ul>
-		<li>A language API</li>
-		<li>Getting Started Guide</li>
-		<li>API Reference Manual</li>
-		<li>Release Notes</li>
-		<li>Sample code.</li>
-	</ul>
-</p>
+openstack.net is an SDK written for the Microsoft .NET platform, designed to enable developers to seamlessly work with
```
Comment (catlook) [inline README.md]: I'd love to see the sentance start with a capital so maybe:  
The SDK openstack.net, written for for the Microsoft .NET platform, is designed to enable ...

**Pull request merged**